### PR TITLE
[Bugfix] Fix support show schemas

### LINKED ISSUE #6769: Superset 1.5 failed to connect Starrocks
### Steps to reproduce the behavior (Required)
1. install superset 1.5
2. use webui to connect starrocks

### Expected behavior (Required)
connect ok
### Real behavior (Required)
2022-05-31 10:19:03,311:ERROR:root:(MySQLdb._exceptions.ProgrammingError) (1064, "no viable alternative at input 'SHOW schemas'")
superset_app            | [SQL: SHOW schemas]
superset_app            | (Background on this error at: http://sqlalche.me/e/13/f405)
superset_app            | Traceback (most recent call last):
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
superset_app            |     self.dialect.do_execute(
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
superset_app            |     cursor.execute(statement, parameters)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
superset_app            |     res = self._query(query)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
superset_app            |     db.query(q)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
superset_app            |     _mysql.connection.query(self, query)
superset_app            | MySQLdb._exceptions.ProgrammingError: (1064, "no viable alternative at input 'SHOW schemas'")
superset_app            | 
superset_app            | The above exception was the direct cause of the following exception:
superset_app            | 
superset_app            | Traceback (most recent call last):
superset_app            |   File "/usr/local/lib/python3.8/site-packages/flask_appbuilder/api/__init__.py", line 85, in wraps
superset_app            |     return f(self, *args, **kwargs)
superset_app            |   File "/app/superset/views/base_api.py", line 112, in wraps
superset_app            |     raise ex
superset_app            |   File "/app/superset/views/base_api.py", line 109, in wraps
superset_app            |     duration, response = time_function(f, self, *args, **kwargs)
superset_app            |   File "/app/superset/utils/core.py", line 1468, in time_function
superset_app            |     response = func(*args, **kwargs)
superset_app            |   File "/app/superset/utils/log.py", line 245, in wrapper
superset_app            |     value = f(*args, **kwargs)
superset_app            |   File "/app/superset/views/base_api.py", line 82, in wraps
superset_app            |     return f(self, *args, **kwargs)
superset_app            |   File "/app/superset/databases/api.py", line 253, in post
superset_app            |     new_model = CreateDatabaseCommand(g.user, item).run()
superset_app            |   File "/app/superset/databases/commands/create.py", line 63, in run
superset_app            |     schemas = database.get_all_schema_names(cache=False)
superset_app            |   File "/app/superset/utils/cache.py", line 119, in wrapped_f
superset_app            |     return f(self, *args, **kwargs)
superset_app            |   File "/app/superset/models/core.py", line 627, in get_all_schema_names
superset_app            |     return self.db_engine_spec.get_schema_names(self.inspector)
superset_app            |   File "/app/superset/db_engine_specs/base.py", line 966, in get_schema_names
superset_app            |     return sorted(inspector.get_schema_names())
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/reflection.py", line 156, in get_schema_names
superset_app            |     return self.dialect.get_schema_names(
superset_app            |   File "<string>", line 2, in get_schema_names
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/reflection.py", line 52, in cache
superset_app            |     ret = fn(self, con, *args, **kw)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/dialects/mysql/base.py", line 2720, in get_schema_names
superset_app            |     rp = connection.execute("SHOW schemas")
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2235, in execute
superset_app            |     return connection.execute(statement, *multiparams, **params)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1003, in execute
superset_app            |     return self._execute_text(object_, multiparams, params)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1172, in _execute_text
superset_app            |     ret = self._execute_context(
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
superset_app            |     self._handle_dbapi_exception(
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
superset_app            |     util.raise_(
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
superset_app            |     raise exception
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
superset_app            |     self.dialect.do_execute(
superset_app            |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
superset_app            |     cursor.execute(statement, parameters)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 206, in execute
superset_app            |     res = self._query(query)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/cursors.py", line 319, in _query
superset_app            |     db.query(q)
superset_app            |   File "/usr/local/lib/python3.8/site-packages/MySQLdb/connections.py", line 254, in query
superset_app            |     _mysql.connection.query(self, query)
superset_app            | sqlalchemy.exc.ProgrammingError: (MySQLdb._exceptions.ProgrammingError) (1064, "no viable alternative at input 'SHOW schemas'")
superset_app            | [SQL: SHOW schemas]
superset_app            | (Background on this error at: http://sqlalche.me/e/13/f405)
superset_app            | 172.18.0.1 - - [31/May/2022:10:19:03 +0000] "POST /api/v1/database/ HTTP/1.1" 500 26 "http://localhost:8088/databaseview/list/?pageIndex=
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.2

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6769

## Problem Summary(Required) ：

we should support show schemas.

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Linkerist — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -300,6 +300,7 @@ classifier
 
 showDatabasesStatement
     : SHOW DATABASES ((FROM | IN) catalog=qualifiedName)? ((LIKE pattern=string) | (WHERE expression))?
+    | SHOW SCHEMAS ((LIKE pattern=string) | (WHERE expression))?
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:303]: Why not SHOW (DATABASES|SCHEMAS) ? then all the syntax can be supported for schemas.

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:303]: Mysql does not normally support the syntax of SHOW SCHEMAS (FROM/IN), and SHOW DATABASE support should be a consideration of compatibility. Does SHOW SCHEMAS will support (FROM/IN) syntax is OK?

**Review by gengjun-git — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:303]: Let's keep compatible with MySQL firstly.

**Review by imay — APPROVED** [review]

**Pull request merged**